File: tests/fixtures/cpp/cross_function_taint.cpp
#include <iostream>
#include <string>

std::string source() {
    return "tainted";
}

std::string passthrough(const std::string& data) {
    return data;
}

void sink(const std::string& value) {
    std::cout << value << std::endl;
}

int main() {
    std::string data = source();
    std::string processed = passthrough(data);
    sink(processed);
    return 0;
}

File: tests/fixtures/typescript/cross_function_taint.ts
function source(): string {
    return "tainted";
}

function passthrough(data: string): string {
    return data;
}

function sink(value: string): void {
    console.log(value);
}

function main(): void {
    const data = source();
    const processed = passthrough(data);
    sink(processed);
}
